ports/cheriot-rtos/hal/uart.cpp

```diff
@@ -0,0 +1,42 @@
+#include <platform-uart.hh>
+
+#include "timer.h"
+
+extern "C" uint8_t uart_get_rx_level(volatile OpenTitanUart<115'200> *block) {
+    return (block->fifoStatus >> 16) & 0xff;
+}
+
+extern "C" uint8_t uart_get_tx_level(volatile OpenTitanUart<115'200> *block) {
```

Comment:
Use the `OpenTitanUart::transmit_fifo_level()` method rather than a hard-coded implementation.

Reply:
done